backend/pkg/modules/internal/analytics_core/init.go

```diff
@@ -72,52 +110,39 @@ func (p *AnalyticsCore) handlerFunction(topic TopicType, partitionID int, msgCha
 		if okTopic && len(topicProccFunctions) > 0 {
 			annotations := make([]interface{}, 0, annotationArrayCapacity)
 			for _, proccFunction := range topicProccFunctions {
-				dataFrames := &ProcFuncDataFrames{
-					dataFrames: nil,
-				}
-				annotations = proccFunction.ProccFunc(topic, dataFrames, partitionID, message, annotations, p)
+				dfShards := p.getDataFramesShardsForFunc(proccFunction, partitionID)
+				annotations = proccFunction.ProccFunc(topic, dfShards, partitionID, message, annotations, p)
 			}
 		}
 	}
 }
 
-//nolint:unparam
-func newModuleRaw() (_ core.Module, err error) {
-	p := &AnalyticsCore{
-		httpHandler:         nil,
-		msgBroker:           nil,
-		accessor:            nil,
-		info:                &core.ModuleInfo{Name: "analytics_core", Description: "analytics_core"},
-		numWorkers:          1,
-		proccFuncRegistered: map[TopicType][]AnalyticsModuleProccFunc{},
-		topics:              make([]TopicType, 0, maxNumTopics),
-	}
-	/* We do not need to expose API at this point
-	handler := &httpHandler{
-		accessor: accessor,
+func (p *AnalyticsCore) getDataFramesShardsForFunc(function AnalyticsModuleProccFunc, partitionID int) map[DataFrameID]dataframe.DataFrame {
+	registeredDfs := p.proccFuncDataframeRegistered[function]
```

Comment:
Need to handle situation where there is no entry for a function

Reply:
also following previous comment need to use here function.GetName() as a key

Reply:
When there is not entry, the nil map is returned. In that case, the returned dfShards is empty, because we didn't enter into the for loop